ompi/mpi/fortran/use-mpi-tkr/mpi-f90-interfaces.h

```diff
@@ -1336,7 +1336,7 @@ interface MPI_Testall
 subroutine MPI_Testall(count, array_of_requests, flag, array_of_statuses, ierror)
   include 'mpif-config.h'
   integer, intent(in) :: count
-  integer, dimension(count), intent(inout) :: array_of_requests
+  integer, dimension(*), intent(inout) :: array_of_requests
   logical, intent(out) :: flag
   integer, dimension(MPI_STATUS_SIZE, count), intent(out) :: array_of_statuses
```

Comment:
It looks like the array_of_statuses argument is also supposed to be `*`:

```suggestion
  integer, dimension(MPI_STATUS_SIZE, *), intent(out) :: array_of_statuses
```

Oddly, our other bindings appear to be correct already.

Reply:
Not a fortran expert - am deferring to @markalle .

Reply:
I think it's still a good change, and I lean toward saying array_of_statuses should have the same change made.  I'm not sure what my original search was, but I suspect I just didn't see that one.

I'm currently confused about the ignore_tkr module which I didn't address here, and which appears to have the same issue.  But I think this was at least a legitimate fix for the tkr module.

Reply:
Committed and squashed.